Cabal/Distribution/PackageDescription/Check.hs

```diff
@@ -185,13 +185,15 @@ checkSanity pkg =
         ++ "Only the non-internal library can have the same name as the package."
 
   , check (not (null duplicateNames)) $
-      PackageBuildImpossible $ "Duplicate sections: " ++ commaSep duplicateNames
+      PackageBuildImpossible $ "Duplicate sections: "
+        ++ commaSep (unUnqualComponentName <$> duplicateNames)
```

Comment:
Can't say I'm the biggest fan of fmap here, when a map makes intent much clearer ;)